src/main/java/com/virtudoc/web/entity/UserAccount.java

```diff
@@ -30,6 +33,28 @@
     @Size(min = 1, max = 32)
     private String username;
 
+    @NotNull
+    @NotEmpty
+    @Column(unique = true)
+    @Email
+    @Size(max = 320) // Specified in RFC 2821 and RFC 3693.
+    private String email;
+
+    @NotNull
+    @NotEmpty
+    private String firstName;
+
+    @NotNull
```

Comment:
All the remaining test errors are from these new fields being invalid in `src/test/java/com/virtudoc/web/UserAccountTest.java` and `src/test/java/com/virtudoc/web/AuthenticationServiceTest.java`. It's up to you if you want to do constraint testing on all these new fields, but you will need to update the UserAccount objects in those tests to have non-null and non-empty values for each of these new fields. Once that's done the rest of those test errors should go away and the builds should come back green.